analyses/ewings-cell-type/02-cellassign-tumor-cell-classification.Rmd

```diff
@@ -0,0 +1,641 @@
+  "python scripts/run-cellassign.py",
+  "--anndata_file '{anndata_file}'",
+  "--output_predictions '{cellassign_predictions_file}'",
+  "--reference '{tumor_markers_mtx_file}'"
+)
+
+system(cellassign_call)
+```
+
+
+We can then read in the predictions file and find the predicted cell type for each cell, either "tumor" or "other". 
+
+```{r}
+# read in predictions file
+tumor_predictions <- readr::read_tsv(cellassign_predictions_file)
+
+# get cell type assignments 
+tumor_celltype_assignments <- get_celltype_assignment(tumor_predictions)
+
+# create a table of assignments 
+table(tumor_celltype_assignments$celltype)
+```
+
+Looking at this, it appears that `CellAssign` is classifying all cells as "other"? 
+I find this hard to believe since this sample was taken directly from a Ewing sarcoma tumor, so there are bound to be tumor cells. 
+
+
+### Run CellAssign with a filtered set of tumor marker genes
+
+In looking at the UMAPs of all marker genes (see `01-marker-gene-tumor-classification.Rmd`) there are quite a few genes that have very low expression across all cells, so perhaps they are adding noise to the model. 
+Although this is a bit circular, we could restrict our list of marker genes to be only those that have a high mean gene expression in our dataset? 
+I would expect that when we do that, we are able to identify more tumor cells that line up with what we would expect based on gene expression patterns. 
+
+```{r}
+# grab the mean gene expression for all marker genes 
+rowdata_df <- rowData(sce) |> 
+  as.data.frame() |> 
+  dplyr::filter(gene_symbol %in% tumor_markers_df$gene_symbol)
+rowdata_df
+```
+
+```{r}
+# what's the mean gene expression of all genes in this sample for reference
+mean(rowData(sce)$mean)
+```
+
+
+There are a few genes that have low gene expression < 1. 
+Let's try and filter out those genes and then use `CellAssign`. 
+
+```{r}
+# get list of marker genes to keep
+filtered_markers <- rowdata_df |> 
+  dplyr::filter(mean > 1) |> 
+  dplyr::pull(gene_ids)
+
+# filter binary mtx 
+filtered_ref_mtx <- tumor_ref_mtx |> 
+  dplyr::filter(ensembl_gene_id %in% filtered_markers)
+
+readr::write_tsv(filtered_ref_mtx, filtered_markers_mtx_file)
+```
+
+
+```{r, eval=FALSE}
+cellassign_call <- glue::glue(
+  "python scripts/run-cellassign.py",
+  "--anndata_file '{anndata_file}'",
+  "--output_predictions '{filtered_predictions_file}'",
+  "--reference '{filtered_markers_mtx_file}'"
+)
+
+system(cellassign_call)
+```
+
+
+Now we can read in the predictions for the filtered marker gene list. 
+
+```{r}
+filtered_predictions <- readr::read_tsv(filtered_predictions_file)
+
+filtered_celltype_assignments <- get_celltype_assignment(filtered_predictions)
+
+table(filtered_celltype_assignments$celltype)
+```
+
+Now we are seeing a few more tumor cells, let's see where they are on the UMAP. 
+
+```{r}
+# first read in tumor normal classifications 
+classifications_df <- readr::read_tsv(classifications_file)
```

Comment:
This is from your manual classifications? If so, I would probably rename this to clarify.